### ISSUE TITLE
Update to use latest Talos provider

### ISSUE THREAD

**Issue by scottslowe** [scottslowe/talos-aws-pulumi#4]
The code is currently using a prerelease/alpha version of a provider using a repository that no longer exists. The code needs to be updated to use the latest provider, which has significant changes from earlier versions.

**Comment by itpick** [comment]
```talosSecrets.ToSecretsTypeOutput().ClientConfiguration(),```
 and
 ``` client.GetConfigurationOutput(ctx, client.GetConfigurationOutputArgs{```
 isn't working in your import update branch.

**Comment by scottslowe** [comment]
>  ```talosSecrets.ToSecretsTypeOutput().ClientConfiguration(),```
> 
>  and
> 
>  ``` client.GetConfigurationOutput(ctx, client.GetConfigurationOutputArgs{```
> 
>  isn't working in your import update branch.

Correct. The provider is currently broken, and it's unclear at this time when (if?) a fix will be available.

**Comment by itpick** [comment]
github.com/pulumiverse/pulumi-talos/sdk v0.2.1-0.20240418152927-2960f4249b3b

somehow this version works.  FYI.

**Comment by itpick** [comment]
And instead use ```ToSecretsOutput().ClientConfiguration(),``` instead of ```ToSecretsTypeOutput().ClientConfiguration(),```

**Comment by scottslowe** [comment]
Need to test 0.4.1, which was released a couple weeks ago. It looks like there were significant changes in 0.3.0 which may have fixed the problem.

**Comment by scottslowe** [comment]
Version 0.6.0 seems to address the problems I was experiencing with earlier releases of the pulumiverse/pulumi-talos provider.
